### PR TITLE
ci: add mergify rules for release-v3.1 branch

### PR DESCRIPTION
**Pull request by Madhu-1**
add mergify rules for backporting and auto-merge of PR's for the release-v3.1 branch.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
This one needs manual merging

File: .mergify.yml [hunk 1]
```diff
@@ -151,6 +151,30 @@ pull_request_rules:
...
+      - base=release-v3.1
+      - label!=DNM
+      - "#changes-requested-reviews-by=0"
+      - "#approved-reviews-by>=1"
+      - "status-success=continuous-integration/travis-ci/pr"
+    actions:
+      merge:
+        method: rebase
+        rebase_fallback: merge
+        strict: smart
```
Comment (nixpanic) [inline .mergify.yml:175]: please see #1365 for a change that would be useful

File: .mergify.yml [hunk 2]
```diff
@@ -151,6 +151,31 @@ pull_request_rules:
...
+          - release-v3.1
+  # automerge backports if CI successfully ran
+  - name: automerge backport release-v3.1
+    conditions:
+      - author=mergify[bot]
+      - base=release-v3.1
+      - label!=DNM
+      - "#changes-requested-reviews-by=0"
+      - "#approved-reviews-by>=1"
+      - "status-success=continuous-integration/travis-ci/pr"
```
Comment (nixpanic) [inline .mergify.yml:170]: should we not require the ci/centos/* status too?

Comment (Madhu-1) [reply .mergify.yml:170]: Yes, I have copy-pasted from the older release will update it

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
@humblec ptal

**Comment by Madhu-1** [comment]
Merging as its need manual merge

**Pull request merged**